tests/test_onnxml_binarizer_converter.py

```diff
@@ -0,0 +1,95 @@
+from hummingbird.ml import convert
+
+if onnx_runtime_installed():
+    import onnxruntime as ort
+if onnx_ml_tools_installed():
+    from onnxmltools import convert_sklearn
+    from onnxmltools.convert.common.data_types import FloatTensorType as FloatTensorType_onnx
+
+
+class TestONNXBinarizer(unittest.TestCase):
+    def _test_binarizer_converter(self, threshold):
+        warnings.filterwarnings("ignore")
+        X = np.array([[1, 2, 3], [4, 3, 0], [0, 1, 4], [0, 5, 6]], dtype=np.float32)
+
+        # Create SKL model for testing
+        model = Binarizer(threshold=threshold)
+        model.fit(X)
+
+        # Create ONNX-ML model
+        onnx_ml_model = convert_sklearn(model, initial_types=[("float_input", FloatTensorType_onnx(X.shape))])
+
+        # Create ONNX model by calling converter
+        onnx_model = convert(onnx_ml_model, "onnx", X)
+
+        # Get the predictions for the ONNX-ML model
+        session = ort.InferenceSession(onnx_ml_model.SerializeToString())
+        output_names = [session.get_outputs()[i].name for i in range(len(session.get_outputs()))]
+        inputs = {session.get_inputs()[0].name: X}
+        onnx_ml_pred = session.run(output_names, inputs)
+
+        # Get the predictions for the ONNX model
+        onnx_pred = onnx_model.transform(X)
+
+        return onnx_ml_pred, onnx_pred
+
+    @unittest.skipIf(
+        not (onnx_ml_tools_installed() and onnx_runtime_installed()), reason="ONNXML test requires ONNX, ORT and ONNXMLTOOLS"
+    )
+    # Check 0.0 threshold
+    def test_binarizer_converter_0thresh(self, rtol=1e-06, atol=1e-06):
+        onnx_ml_pred, onnx_pred = self._test_binarizer_converter(0.0)
+
+        np.testing.assert_allclose(onnx_ml_pred, onnx_pred, rtol=rtol, atol=atol)
+
+    @unittest.skipIf(
+        not (onnx_ml_tools_installed() and onnx_runtime_installed()), reason="ONNXML test requires ONNX, ORT and ONNXMLTOOLS"
+    )
+    # Check positive threshold
+    def test_binarizer_converter_posthresh(self, rtol=1e-06, atol=1e-06):
+        onnx_ml_pred, onnx_pred = self._test_binarizer_converter(2.0)
+
+        # Check that predicted values match
+        np.testing.assert_allclose(onnx_ml_pred, onnx_pred, rtol=rtol, atol=atol)
+
+    @unittest.skipIf(
+        not (onnx_ml_tools_installed() and onnx_runtime_installed()), reason="ONNXML test requires ONNX, ORT and ONNXMLTOOLS"
+    )
+    # Check neg threshold
+    def test_binarizer_converter_negthresh(self, rtol=1e-06, atol=1e-06):
+        onnx_ml_pred, onnx_pred = self._test_binarizer_converter(-2.0)
+
+        # Check that predicted values match
+        np.testing.assert_allclose(onnx_ml_pred, onnx_pred, rtol=rtol, atol=atol)
+
+    @unittest.skipIf(
+        not (onnx_ml_tools_installed() and onnx_runtime_installed()), reason="ONNXML test requires ONNX, ORT and ONNXMLTOOLS"
+    )
+    # if the model is corrupt, we should get a RuntimeError
+    def test_onnx_binarizer_converter_raises_rt(self):
```

Comment:
Should we have this test in some other base test? I think it is independent from binarizer, no?

Reply:
This hits line 36 of `binarizer.py`.  It's specific to `Binarizer` in that it makes sure the threshold is set on lines 31-33.  (Mostly, I wanted 100% coverage :-D )

Q- Is this exception case realistic?  Alternatively, I could remove lines 35-36 of`binarizer.py` since I am not sure they can actually ever be reached unless a user specifically corrupts the object like I did for the  test. :)

Reply:
I don't know! I would say to leave it, better be conservative.